plugins/modules/ndo_tenant_igmp_interface_policy.py

```diff
@@ -0,0 +1,535 @@
+    type: str
+    aliases: [ static_report_route_map_for_multicast ]
+  static_report_route_map:
+    description:
+    - The Route Map Policy for Multicast.
+    - This parameter can be used instead of O(static_report_route_map_uuid).
+    - If both parameter are used, O(static_report_route_map) will be ignored.
+    - Providing an empty map will remove the O(static_report_route_map={}) from IGMP Interface Policy.
+    type: dict
+    suboptions:
+      name:
+        description:
+        - The name of the Route Map Policy for Multicast.
+        type: str
+    aliases: [ static_report_route_map_policy, static_report_route_map_policy_multicast ]
+  maximum_multicast_entries:
+    description:
+    - The maximum multicast entries value.
+    - Defaults to 4294967295 when unset during creation.
+    - The valid range is from 1 to 4294967295.
+    - This parameter is only applicable when the O(state_limit_route_map_uuid) or O(state_limit_route_map) is not empty.
+    type: int
+  reserved_multicast_entries:
+    description:
+    - The reserved multicast entries value.
+    - Defaults to 0 when unset during creation.
+    - The valid range is from 0 to 4294967295.
+    type: int
+  state:
+    description:
+    - Use C(absent) for removing.
+    - Use C(query) for listing an object or multiple objects.
+    - Use C(present) for creating or updating.
+    type: str
+    choices: [ absent, query, present ]
+    default: query
+notes:
+- The O(template) must exist before using this module in your playbook.
+  Use M(cisco.mso.ndo_template) to create the Tenant template.
+- The O(state_limit_route_map_uuid) or O(state_limit_route_map) requires Route Map Policy for Multicast.
+  The O(report_policy_route_map_uuid) or O(report_policy_route_map) requires Route Map Policy for Multicast.
+  The O(static_report_route_map_uuid) or O(static_report_route_map) requires Route Map Policy for Multicast.
+  Use M(cisco.mso.ndo_route_map_policy_multicast) to create the Route Map Policy for Multicast.
+seealso:
+- module: cisco.mso.ndo_template
+- module: cisco.mso.ndo_route_map_policy_multicast
+extends_documentation_fragment: cisco.mso.modules
+"""
+
+EXAMPLES = r"""
+- name: Create an IGMP Interface Policy
+  cisco.mso.ndo_tenant_igmp_interface_policy:
+    template: ansible_tenant_template
+    name: test_igmp_interface_policy
+    description: Test Description
+    version3_asm: enabled
+    fast_leave: enabled
+    report_link_local_groups: enabled
+    igmp_version: v3
+    group_timeout: 260
+    query_interval: 125
+    query_response_interval: 10
+    last_member_count: 2
+    last_member_response_time: 1
+    startup_query_count: 2
+    startup_query_interval: 31
+    querier_timeout: 255
+    robustness_variable: 2
+    state_limit_route_map_uuid: route_map_policy_for_multicast_uuid
+    report_policy_route_map_uuid: route_map_policy_for_multicast_uuid
```

Comment:
when providing uuid, should we provide reference or actual uuid format instead of a string?

Reply:
Added UUID sting in one and reference in another.